### ISSUE TITLE
Checks that require network connection: check if commit belongs to any branch on that repo

### ISSUE THREAD

**Issue by SuperSandro2000** [jtojnar/nixpkgs-hammering#65]
Based on https://github.com/NixOS/nixpkgs/issues/20836#issuecomment-767380571 I would like to request a check that checks fetchFromGitHub for commits that are from a fork and may be malicious.

**Comment by rmcgibbo** [comment]
This is a good idea, and I think it's not too hard do to after merging #56.

**Comment by rmcgibbo** [comment]
Ideas for how to do this (more complicated that I previously realized):
1. Try to do it statically in `rnix`. Bad idea, because the arguments passed to fetchFromGitHub aren't fully static.
2. Use existing overlays / `reports`. Complicated by the fact that there can be a long chain of derivations, where we're hammering attr `A`, but `A` depends on `B` depends on `C`, and `B` and `C` are not themselves addressable top-level nixpkgs attributes, but still might contain a `fetchFromGitHub` that we want to audit. In this case, it's hard to attach the report to the reports for attr `A`.
3. Override `fetchurl` (or similar) with a traced version:

```
overlay_expressions.append(textwrap.dedent(
    f"""
    (self: super: {{
      fetchurl = let
        originalOutput = args: builtins.trace (builtins.toJSON args) (super.fetchurl args);            
      in
        originalOutput;
    }})
    """
))
```
and then read the traced data out of stdout when we execute the overlays. With #56, since the calculation of the drvs is forced, this actually works. But it's ugly and mixes up the trace information we want with any other trace information, since we're just parsing stdout.
4. Morally do (3), but use either a new plugin-created primop, builtins.importNative, or builtins.exec with `--option allow-unsafe-native-code-during-evaluation true` in place of `builtins.trace`, so that our information about what arguments were passed to fetchurl calls during evaluation is not intermingled with any other trace information.

I tried (3), and it does work, but I think (4) might be the way to go.

**Comment by rmcgibbo** [comment]
Here's an example of idea (4): https://github.com/rmcgibbo/nixpkgs-hammering/commit/0ca4bdc77de1e4770c729f202b888cdf20be2c77

**Comment by jtojnar** [comment]
That plug-in looks good, though are not file descriptors per-process?

**Comment by rmcgibbo** [comment]
Yeah, file descriptors are per-process. What's the implication of that? Presumably nothing else should be writing to that file descriptor because nix gives you no interface to write unsafe IO like that, and we can open multiple different `traceFds` to write different types of data streams, if we need to.
